zim/applications.py

```diff
@@ -311,7 +311,7 @@ def run(self, args=None, cwd=None):
 			try:
 				p = subprocess.Popen(argv,
 					cwd=cwd,
-					stdout=open(os.devnull, 'w'),
+					stdout=subprocess.DEVNULL,
 					stderr=subprocess.PIPE,
 					bufsize=4096,
 					close_fds=True
```

Comment:
`close_fds` defaults to `True` [from 3.2 onwards](https://docs.python.org/3/library/subprocess.html) while `subprocess.DEVNULL` was introduced in 3.3. If we're changing the code for modernization's sake, it should be done consistently on a given level (for example, a whole function/call at a time) or else we'll end up with a patchwork of modern and legacy code.

Reply:
@jaap-karssenberg sorry for the hasty initial review; do you think we should have some guidelines on refactoring? Given the age of the code base, we do have some technical debt on tab, some of which can be paid now that we've bumped to 3.6+.